cabal-install/solver-dsl/UnitTests/Distribution/Solver/Modular/DSL.hs

```diff
@@ -719,13 +720,13 @@ extractInstallPlan :: CI.SolverInstallPlan.SolverInstallPlan
 extractInstallPlan = catMaybes . map confPkg . CI.SolverInstallPlan.toList
   where
     confPkg :: CI.SolverInstallPlan.SolverPlanPackage -> Maybe (String, Int)
-    confPkg (CI.SolverInstallPlan.Configured pkg) = Just $ srcPkg pkg
+    confPkg (CI.SolverInstallPlan.Configured pkg) = srcPkg pkg
     confPkg _                               = Nothing
 
-    srcPkg :: SolverPackage UnresolvedPkgLoc -> (String, Int)
+    srcPkg :: SolverPackage UnresolvedPkgLoc -> Maybe (String, Int)
     srcPkg cpkg =
       let C.PackageIdentifier pn ver = packageInfoId (solverPkgSource cpkg)
-      in (C.unPackageName pn, head (C.versionNumbers ver))
+      in (\vn -> (C.unPackageName pn, vn)) <$> safeHead (C.versionNumbers ver)
```

Comment:
Oh, this is subtle.

Almost asks for having two `Version` variants, the one which can be `nullVersion`, and one which cannot. But that's a huge change. cc @hvr

Reply:
In terms of the diff here, is the change fine?  My interpretation is that when extracting the install plans it would skip any package with the null version, rather than blowing up.  Does that sound like a change in the right direction?

Reply:
Your current change is fine.

There shouldn't be any `nullVersion` there, but we cannot be sure without a very big refactoring (which payoff is questionable).